### PR TITLE
Add Exoscale provider

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Add Exoscale provider to ignition

Fetch directly user-data server to retrieve ignition user configuration

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -6,6 +6,7 @@ github.com/coreos/go-semver v0.3.0 h1:wkHLiw0WNATZnSG7epLsujiMCgPAc9xhjJ4tgnAxmf
 github.com/coreos/go-semver v0.3.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
 github.com/coreos/go-systemd v0.0.0-20181031085051-9002847aa142 h1:3jFq2xL4ZajGK4aZY8jz+DAF0FHjI51BXjjSwCzS1Dk=
 github.com/coreos/go-systemd v0.0.0-20181031085051-9002847aa142/go.mod h1:F5haX7vjVVG0kc13fIWeqUViNPyEJxv/OmvnBo0Yme4=
+github.com/coreos/ignition v0.35.0 h1:UFodoYq1mOPrbEjtxIsZbThcDyQwAI1owczRDqWmKkQ=
```
Comment (jlebon) [inline go.sum]: This looks suspect... how did that end up here?

File: internal/providers/exoscale/exoscale.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+var (
+	userdataURL = url.URL{
+		Scheme: "http",
+		Host:   "169.254.169.254",
+		Path:   "1.0/user-data",
+	}
+)
+
+func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```
Comment (jlebon) [inline internal/providers/exoscale/exoscale.go]: Hmm, I think we can just use `f.fetchToBuffer` directly here, right? See e.g. `aws.go`. IIRC, the client the fetcher uses already has a timeout (which can be overridden via the timeout Ignition config knob).

File: internal/providers/exoscale/exoscale.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2016 CoreOS, Inc.
```
Comment (arithx) [inline internal/providers/exoscale/exoscale.go]: s,2016,2020

File: go.sum (hunk 1 above)
Comment (arithx) [reply go.sum]: Definitely shouldn't be added there. Could have happened if the code initially imported a new path that didn't have the `/v2` included and a build was performed (go mod tries to be smart about these things).

File: internal/providers/exoscale/exoscale.go (hunk 3 above)
Comment (pierre-emmanuelJ) [reply internal/providers/exoscale/exoscale.go]: thanks

File: internal/providers/exoscale/exoscale.go (hunk 2 above)
Comment (pierre-emmanuelJ) [reply internal/providers/exoscale/exoscale.go]: Okay thanks, so I have simplified the code. I'll check timeout ignition knob. Have you got a link to that?

**Comment by pierre-emmanuelJ** [comment]
I have an issue if the result of the query is empty and 200 http ok (no user-data provided by the user, empty)

I'm blocked at startup:
image

**Comment by lucab** [comment]
@pierre-emmanuelJ that sounds like the machine booted properly, but the prompt is on a different (serial) console. Even without credentials to log in, are you able to reach the SSH server at the network level at that point?

**Comment by pierre-emmanuelJ** [comment]
> @pierre-emmanuelJ that sounds like the machine booted properly, but the prompt is on a different (serial) console. Even without credentials to log in, are you able to reach the SSH server at the network level at that point?

Yes it seems to be that but I can not reach the SSH server

File: go.sum (hunk 1 above)
Comment (pierre-emmanuelJ) [reply go.sum]: I fixed is it ok for you?

**Comment by pierre-emmanuelJ** [comment]
so I do not have the issue anymore for empty ignition config

**Comment by lucab** [comment]
@pierre-emmanuelJ can you please just squash all the commits into a single one?

**Comment by pierre-emmanuelJ** [comment]
Squash done thanks!

**Comment by pierre-emmanuelJ** [comment]
Thanks for reviews

**Pull request merged**